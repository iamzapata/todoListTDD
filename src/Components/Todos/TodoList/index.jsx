import React, { PureComponent } from "react"
import { arrayOf, shape, string, number, func } from "prop-types"
import Todo from "../Todo"

class TodoList extends PureComponent {
  render() {
    const { todos, toggleTodo } = this.props
    return (
      <ul className="TodosList">
        {todos.map(td => (
          <li key={td.id}>
            <Todo todo={td} toggleTodo={toggleTodo} />
          </li>
        ))}
      </ul>
    )
  }
}

TodoList.propTypes = {
  todos: arrayOf(
    shape({
      text: string,
      id: number
    })
  ).isRequired,
  toggleTodo: func.isRequired
}

export default TodoList
