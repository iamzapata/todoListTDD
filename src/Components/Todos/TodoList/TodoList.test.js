import React from "react"
import { render, cleanup } from "react-testing-library"
import TodoList from "./index.jsx"

describe("<TodoList />", () => {
  afterEach(cleanup)
  const createProps = () => ({
    todos: [
      {
        id: 1,
        text: "Wash dishes",
        completed: false
      },
      {
        id: 2,
        text: "Take out trash",
        completed: false
      }
    ],
    toggleTodo: jest.fn()
  })
  it("Should render self", () => {
    const { container } = render(<TodoList {...createProps()} />)
    expect(container.firstChild).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })

  it("Should render a top level ul", () => {
    const { container } = render(<TodoList {...createProps()} />)
    const ulElement = container.getElementsByTagName("ul")

    expect(ulElement.length).toBe(1)
  })

  it("Should have a TodosList className", () => {
    const { container } = render(<TodoList {...createProps()} />)
    const classNameSelection = container.querySelectorAll(".TodosList")

    expect(classNameSelection.length).toBe(1)
  })

  it("Should render a list of todos", () => {
    const { container } = render(<TodoList {...createProps()} />)
    const children = container.querySelectorAll("li")

    expect(children.length).toEqual(2)
  })

  it("Should render a list of <Todo /> components", () => {
    const { container } = render(<TodoList {...createProps()} />)
    const todoInputChildren = container.querySelectorAll("li .Todo")

    expect(todoInputChildren.length).toEqual(2)
  })
})
