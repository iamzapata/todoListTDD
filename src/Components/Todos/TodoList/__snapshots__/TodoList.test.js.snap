// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> Should render self 1`] = `
<ul
  class="TodosList"
>
  <li>
    <span
      class="Todo"
    >
      Wash dishes
      <input
        type="checkbox"
      />
    </span>
  </li>
  <li>
    <span
      class="Todo"
    >
      Take out trash
      <input
        type="checkbox"
      />
    </span>
  </li>
</ul>
`;
